src/ui-test/uitest_runner.ts

```diff
@@ -16,19 +16,21 @@
  */
 'use strict';
 
-import * as path from 'path';
 import * as fs from 'fs';
-import { ExTester } from 'vscode-extension-tester';
-import { ReleaseQuality } from 'vscode-extension-tester/out/util/codeUtil';
+import { variables } from './variables';
+import { Repeat, ExTester, ReleaseQuality } from 'vscode-uitests-tooling';
 
 export const storageFolder = 'test-resources';
 const releaseType: ReleaseQuality = process.env.CODE_TYPE === 'insider' ? ReleaseQuality.Insider : ReleaseQuality.Stable;
-export const projectPath = path.resolve(__dirname, '..', '..');
-const extensionFolder = path.join(projectPath, '.test-extensions');
 
 async function main(): Promise<void> {
-	const tester = new ExTester(storageFolder, releaseType, extensionFolder);
-	await tester.setupAndRunTests('out/ui-test/tests/*.test.js',
+    process.on('exit', cleanup);
```

Comment:
actually I don't like we are again changing this runner only for DAP on top of as part of adding a new UI test...

is are these changes (`package.json`,`uitest_runner`, `src/ui-testvariables/*`) necessary? I agree it is an improvement but I would expect as separate PR which will be also presented for other projects to still have unification in place and avoid this kind of diversification

Reply:
> actually I don't like we are again changing this runner only for DAP on top of as part of adding a new UI test...

What is a point of creating PR for each extension with those runner changes? Firstly, runner unification is more complex than ui-test one, and it was perfectly showcased by this interaction. In attempt to reach unification, some functionality was removed which I used for debugging purposes, that returns in this PR (for some time it was bearable to remove new functionality before commit and then add it, but it is redundant task to do). The problem is, it requires both sides to pay attention which sometimes it is hard to do (committer must ensure that core functionality is not lost and the other side must have time to perform thorough review, therefore it is time consuming for both sides). Secondly, mandating to use ui-test npm script and the same file tree is enough in my opinion. And at last, if a new feature is really useful, it will eventually be adopted by other code owners.

> is are these changes (`package.json`,`uitest_runner`, `src/ui-testvariables/*`) necessary? I agree it is an improvement but I would expect as separate PR which will be also presented for other projects to still have unification in place and avoid this kind of diversification

Variables ... yes, or do you want me to declare them in utils.ts or hardcode them?

Just my 2 cents. I can implement both variants. It is your choice, as this change is not going to affect me in long term.

Reply:
so why we were doing some unification at all in first place if you are saying this unification is more complex and it is on each code owner to adopt?.. this will lead to different runners and approaches --> means back to beginning..

I do not think that runner modifications are that much often to not have unified them to some state which we discuss and approve and it will suits for every component...

I see your point and I see you probably missing something after modifications which was made during unification but why we did not talked about this missing during time we were doing the migration?

I really believe that from this kind of improvements which affecting some unified part of codebase can mine more than one code owner and its a pity to do not discuss it separately and potentially to contribute also to other components

Reply:
regarding to variables... I don't just probably do not completely understand of reexporting across three new files because of 4 variables..

because this mention in description `Please note, as part of this PR I have experimented with variables structure, because I believe single utils.ts is not optimal as long term solution.` I believed that it is something we know already and that why are trying to bring back and improve vscode-uitests-tooling, or am I wrong?

Reply:
The point of having multiple variable files is better organisation. The whole idea resolves around namespaces instead 1  generic file. It is possible to include them in vscode-uitests-tooling, however it might be annoying if you want to change any variable temporarily.

Reply:
About reunification, not a big deal, you can break unification whenever it is needed. Therefore, I will revert the runner except rmSync, as it wrongly assumes instructions above never throw errors. Despite my point above, I am going to revert variable changes as well, since these changes can be reimplemented easily.